cli/command/service/progress/progress.go

```diff
@@ -388,7 +426,23 @@ func (u *globalProgressUpdater) update(service swarm.Service, tasks []swarm.Task
 
 	for _, task := range tasksByNode {
 		if node, nodeActive := activeNodes[task.NodeID]; nodeActive {
-			if !u.done && nodeCount <= maxProgressBars {
+			if !terminalState(task.DesiredState) && task.Status.State == swarm.TaskStateRunning {
+				running++
+			}
+
+			if u.done || nodeCount > maxProgressBars {
```

Comment:
same question about `nodeCount > maxProgressBars` not changing value within this for loop.

And also splitting this out into a new function (which should remove the need for nolint, and help testing).